modules/related-posts/class.related-posts-customize.php

```diff
@@ -0,0 +1,202 @@
+	 */
+	function customize_register( $wp_customize ) {
+
+		$wp_customize->add_section( $this->prefix,
+			array(
+				'title' 	  => esc_html__( 'Related Posts', 'jetpack' ),
+				'description' => esc_html__( 'Customize related post elements.', 'jetpack' ),
+				'capability'  => 'edit_theme_options',
+				'priority' 	  => 89,
+
+			)
+		);
+
+		$selective_options = array();
+
+		foreach ( $this->get_options( $wp_customize ) as $key => $field ) {
+			$selective_options[] = "$this->prefix[$key]";
+			$wp_customize->add_setting( "$this->prefix[$key]",
+				array(
+					'default' 	 => isset( $field['default'] ) ? $field['default'] : '',
+					'type' 		 => isset( $field['setting_type'] ) ? $field['setting_type'] : 'option',
+					'capability' => isset( $field['capability'] ) ? $field['capability'] : 'edit_theme_options',
+					'transport'  => isset( $field['transport'] ) ? $field['transport'] : 'postMessage',
+				)
+			);
+			$control_settings = array(
+				'label' 	  => isset( $field['label'] ) ? $field['label'] : '',
+				'description' => isset( $field['description'] ) ? $field['description'] : '',
+				'settings' 	  => "$this->prefix[$key]",
+				'type' 	      => isset( $field['control_type'] ) ? $field['control_type'] : 'text',
+				'section' 	  => $this->prefix,
+				'priority' 	  => 10,
+				'active_callback' => __CLASS__ . '::is_single',
+			);
+			switch ( $field['control_type'] ) {
+				case 'text':
+				case 'checkbox':
+				default:
+					$wp_customize->add_control( new WP_Customize_Control( $wp_customize, $key, $control_settings ) );
+					break;
+				case 'select':
+					if ( isset( $field['choices'] ) ) {
+						$control_settings['choices'] = $field['choices'];
+						$wp_customize->add_control( new WP_Customize_Control( $wp_customize, $key, $control_settings ) );
+					}
+					break;
+				case 'message':
+					$control_settings['active_callback'] = __CLASS__ . '::is_not_single';
+					$wp_customize->add_control( new Jetpack_Message_Control( $wp_customize, $key, $control_settings ) );
+					break;
+			}
+		}
+
+		// If selective refresh is available, implement it.
+		if ( isset( $wp_customize->selective_refresh ) ) {
+			$wp_customize->selective_refresh->add_partial( "$this->prefix", array(
+				'selector'            => '.jp-relatedposts',
+				'settings'            => $selective_options,
+				'render_callback'     => __CLASS__ . '::render_callback',
+				'container_inclusive' => false,
+			) );
+		}
+
+	}
+
+	/**
+	 * Callback that outputs the headline based on user choice.
+	 *
+	 * @since 4.2.0
+	 */
+	public static function render_callback() {
+		echo Jetpack_RelatedPosts::init()->get_headline();
+	}
+
+	/**
+	 * Check that we're in a single post view.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @return bool
+	 */
+	public static function is_single() {
+		return is_single();
+	}
+
+	/**
+	 * Check that we're not in a single post view.
+	 *
+	 * @since 4.2.0
+	 *
+	 * @return bool
+	 */
+	public static function is_not_single() {
```

Comment:
Same here, do we need this? Also, if we have the `is_single()` method, do we need a whole other method to negate that?